### PR TITLE
Consumer Api: Deletion of Files, Tokens and Relationship Templates

### PR DESCRIPTION
**Pull request by MH321Productions**
# Readiness checklist

-   [ ] I added/updated unit tests.
-   [x] I added/updated integration tests.
-   [x] I ensured that the PR title is good enough for the changelog.
-   [x] I labeled the PR.

### CODE REVIEW

**Review by tnotheis — CHANGES_REQUESTED** [review]

File: Applications/ConsumerApi/test/ConsumerApi.Tests.Integration/Features/RelationshipTemplates/{id}/DELETE.feature [hunk 1]
```diff
@@ -4,9 +4,16 @@ Feature: DELETE /RelationshipTemplates/{id}
...
+        And an active Relationship r between i1 and i2 with template t
+        When i1 sends a DELETE request to the /RelationshipTemplates/t.Id endpoint
+        Then the Relationship r still exists
+        And the Relationship r does not have a relationship template
+
+    Scenario: Deleting a template actually removes it
+        Given Identity i
+        And a Relationship Template t created by i
+        When i sends a DELETE request to the /RelationshipTemplates/t.Id endpoint
+        And i sends a GET request to the /RelationshipTemplates/t.Id endpoint with password "-"
```
[the remark this answers is not in the document]
Comment (MH321Productions) [reply Applications/ConsumerApi/test/ConsumerApi.Tests.Integration/Features/RelationshipTemplates/{id}/DELETE.feature:18]: The "-" means that no password is used (I didn't want to write the same code again)

File: Applications/ConsumerApi/test/ConsumerApi.Tests.Integration/Features/Tokens/{id}/DELETE.feature [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿@Integration
+Feature: DELETE /Tokens/{id}
+
+User deletes a Token
+
+    Scenario: Deleting a Token actually removes it
+        Given Identity i
+        And Token t created by i with password "-" and forIdentity -
```
[the remark this answers is not in the document]
Comment (MH321Productions) [reply Applications/ConsumerApi/test/ConsumerApi.Tests.Integration/Features/Tokens/{id}/DELETE.feature:8]: See above

File: Applications/ConsumerApi/test/ConsumerApi.Tests.Integration/Features/RelationshipTemplates/{id}/DELETE.feature (hunk 1 above)
[the remark this answers is not in the document]
Comment (tnotheis) [reply Applications/ConsumerApi/test/ConsumerApi.Tests.Integration/Features/RelationshipTemplates/{id}/DELETE.feature:18]: In my opinion having tests that have better readability is worth more than saving a little bit of code.

The step definitions with the `-` were written for tests where we had a table with input data. In that case it was handy to have something that meant "no password". But in this case you might as well just skip the whole password part.

File: Applications/ConsumerApi/test/ConsumerApi.Tests.Integration/StepDefinitions/TokensStepDefinitions.cs [hunk 3]
```diff
@@ -121,6 +132,20 @@ public async Task WhenIdentitySendsAGetRequestToTheTokensIdEndpointWithPassword(
...
+        var isAuthenticated = identityName != "-";
+
+        var client = isAuthenticated ? _clientPool.FirstForIdentityName(identityName) : _clientPool.Anonymous;
+        var tokenId = _tokensContext.CreateTokenResponses[tokenName].Id;
+
+        if (isAuthenticated)
+            _responseContext.WhenResponse = await client.Tokens.GetToken(tokenId);
+        else
+            _responseContext.WhenResponse = await client.Tokens.GetTokenUnauthenticated(tokenId);
+    }
```
Comment (tnotheis) [inline Applications/ConsumerApi/test/ConsumerApi.Tests.Integration/StepDefinitions/TokensStepDefinitions.cs:147]: No need to support the anonymous case here.

```suggestion
    [When($@"{RegexFor.SINGLE_THING} sends a GET request to the /Tokens/{RegexFor.SINGLE_THING}.Id endpoint")]
    public async Task WhenIdentitySendsAGetRequestToTheTokensIdEndpoint(string identityName, string tokenName)
    {
        var client = _clientPool.FirstForIdentityName(identityName);
        var tokenId = _tokensContext.CreateTokenResponses[tokenName].Id;
        _responseContext.WhenResponse = await client.Tokens.GetToken(tokenId);
    }
```

**Review by tnotheis — CHANGES_REQUESTED** [review]

**Review by tnotheis — APPROVED** [review]

**Pull request merged**